Sources/Tools/plShaderAssembler/main.cpp

```diff
@@ -162,78 +175,83 @@ static void IAssShader(const plDXShaderAssembler& ass, const char* name)
         fputs("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n", stderr);
     }
 }
-
+// Just copy in the enum and use Replace on
+// vs_ => "vs_
+// ps_ => "ps_
+// , => ",
+static constexpr std::string_view kEnumNames[] = {
+    "vs_WaveFixedFin6",
+    "ps_WaveFixed",
+    "vs_CompCosines",
+    "ps_CompCosines",
+    "vs_ShoreLeave6",
+    "ps_ShoreLeave6",
+    "vs_WaveRip",
+    "ps_WaveRip",
+    "vs_WaveDec1Lay",
+    "vs_WaveDec2Lay11",
+    "vs_WaveDec2Lay12",
+    "vs_WaveDecEnv",
+    "ps_CbaseAbase",
+    "ps_CalphaAbase",
+    "ps_CalphaAMult",
+    "ps_CalphaAadd",
+    "ps_CaddAbase",
+    "ps_CaddAMult",
+    "ps_CaddAAdd",
+    "ps_CmultAbase",
+    "ps_CmultAMult",
+    "ps_CmultAAdd",
+    "ps_WaveDecEnv",
+    "vs_WaveGraph2",
+    "ps_WaveGraph",
+    "vs_WaveGridFin",
+    "ps_WaveGrid",
+    "vs_BiasNormals",
+    "ps_BiasNormals",
+    "vs_ShoreLeave7",
+    "vs_WaveRip7",
+    "ps_MoreCosines",
+    "vs_WaveDec1Lay_7",
+    "vs_WaveDec2Lay11_7",
+    "vs_WaveDec2Lay12_7",
+    "vs_WaveDecEnv_7",
+    "vs_WaveFixedFin7",
+    "vs_GrassShader",
+    "ps_GrassShader"
+};
 int main(int argc, char* argv[])
 {
-    if( argc < 2 )
+    std::vector<std::string_view> allArgs(argv, argv + argc);
+    //remove command
+    allArgs.erase(allArgs.begin());
+
+    if( allArgs.empty() )
```

Comment:
This honestly just makes things seem more complicated to me...

Reply:
renamed it in a later commit, but its use is so that the argument array is of the same type as the constexpr